src/core/src/main/java/org/apache/jmeter/util/keystore/JmeterKeyStore.java

```diff
@@ -184,37 +170,27 @@ private boolean isIndexInConfiguredRange(int index) {
     /**
      * Get the ordered certificate chain for a specific alias.
      *
-     * @param alias
-     *            the alias for which the certificate chain should be given
-     * @return the certificate chain for the alias
-     * @throws IllegalArgumentException
-     *             if no chain could be found for the alias
+     * @param alias the alias for which the certificate chain should be given
+     * @return the certificate chain for the alias or null if not found
+     * @see javax.net.ssl.X509KeyManager#getCertificateChain(String)
      */
     public X509Certificate[] getCertificateChain(String alias) {
-        X509Certificate[] result = this.certsByAlias.get(alias);
-        if(result != null) {
-            return result;
-        }
-        // API expects null not empty array, see http://docs.oracle.com/javase/7/docs/api/javax/net/ssl/X509KeyManager.html
-        throw new IllegalArgumentException("No certificate found for alias:'"+alias+"'");
+        return this.certsByAlias.get(alias);
```

Comment:
Hello @ham1 , is this change in behaviour needed ?
Thanks

Reply:
This change aligns to the comments; i.e. it will return null if not found as per the comment.
The current behaviour is to throw an `IllegalArgumentException` which is different from the documentation.

Reply:
I'm not quite sure what you mean but I'm happy to revert as it might indeed provide a better error msg in the GUI than to have a NPE later.